### DISCUSSION TITLE
[Feature] Flink Connector support pendingRecords metric

### DISCUSSION THREAD

**Discussion by luoyuxia** [apache/fluss#91] [General]
### Search before asking

- [X] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.

### Motivation

Although we has coverd most of standard metrics defined in [FLIP-33: Standardize Connector Metrics](https://cwiki.apache.org/confluence/display/FLINK/FLIP-33%3A+Standardize+Connector+Metrics), we still miss the metrics of `pendingRecords` which help us to trace data backlog.

### Solution

pending records  = `latest log offset - fetched log offset`. 

For snapshot read phase of primary key table, it should be total records in kv snapshot - records emited, but to simplify, we can just trace the phase of incremental reading which read from change log.

### Willingness to contribute

- [ ] I'm willing to submit a PR!

**Comment by HZY-Wade** [comment]
I would like to contribute this issue, please assign it to me

**Comment by wuchong** [comment]
convert to issue #138
